src/gui/editors/AutomationEditor.cpp

```diff
@@ -694,32 +686,78 @@ void AutomationEditor::mouseReleaseEvent(QMouseEvent * mouseEvent )
 
 
 
-void AutomationEditor::removePoints( int x0, int x1 )
+/* @brief Removes all automation nodes between the given ticks
+ * @param Int first tick of the range
+ * @param Int second tick of the range
+ */
+void AutomationEditor::removeNodes(int tick0, int tick1)
 {
-	int deltax = qAbs( x1 - x0 );
-	int x = x0;
-	int xstep;
+	QMutexLocker m(&m_patternEditorMutex);
 
-	if( deltax < 1 )
+	if (tick0 == tick1)
 	{
 		return;
 	}
 
-	if( x0 < x1 )
+	MidiTime start = MidiTime(qMin(tick0, tick1));
+	MidiTime end = MidiTime(qMax(tick0, tick1));
+
+	timeMap & tm = m_pattern->getTimeMap();
+	timeMap::iterator it = tm.lowerBound(start);
+
+	// Make a list of MidiTimes with nodes to be removed
+	// because we can't simply remove the nodes from
+	// the timeMap while we are iterating it.
+	QVector<MidiTime> nodesToRemove;
+
+	while (it != tm.end())
 	{
-		xstep = 1;
+		if (it.key() > end)
+		{
+			break;
+		}
+
+		nodesToRemove.append(it.key());
+		++it;
 	}
-	else
+
+	for (int c = 0; c < nodesToRemove.size(); ++c)
 	{
-		xstep = -1;
+		m_pattern->removeValue(nodesToRemove[c]);
 	}
+}
 
-	int i = 0;
-	while( i <= deltax )
+
+
+
+/* @brief Resets the outValues of all automation nodes between the given ticks
+ * @param Int first tick of the range
+ * @param Int second tick of the range
+ */
+void AutomationEditor::resetNodes(int tick0, int tick1)
+{
+	QMutexLocker m(&m_patternEditorMutex);
+
+	if (tick0 == tick1)
```

Comment:
Short if, we could combine lines:
```suggestion
	if (tick0 == tick1) { return; }
```